src/main/groovy/lang/GroovyCodeSource.java

```diff
@@ -164,7 +164,7 @@ public GroovyCodeSource(URI uri) throws IOException {
         this(uri.toURL());
     }
 
-    public GroovyCodeSource(URL url) throws IOException {
```

Comment:
I only just spotted this now. While "unused" in the sense that this method didn't throw this exception anymore (I presume it did in the past), removing the exception breaks contract/source code compatibility. :-( I know it is a result of how checked exceptions are handled on the JVM but we should have not done this in a point release - all good to know in hindsight. Anyway, it isn't clear that reverting after a year is going to help things so I'm thinking of just documenting for now with a Jira issue. Apologies if this has already been raised elsewhere.